subsys/zigbee/lib/zigbee_app_utils/zigbee_app_utils.c

```diff
@@ -32,24 +32,38 @@
 
 #define IEEE_ADDR_BUF_SIZE       17
 
+#define FACTORY_RESET_PROBE_TIME K_SECONDS(1)
+
 LOG_MODULE_REGISTER(zigbee_app_utils, CONFIG_ZIGBEE_APP_UTILS_LOG_LEVEL);
 
 /* Rejoin-procedure related variables. */
-static bool           stack_initialised;
-static bool           is_rejoin_procedure_started;
-static bool           is_rejoin_stop_requested;
-static bool           is_rejoin_in_progress;
-static uint8_t           rejoin_attempt_cnt;
+static bool stack_initialised;
+static bool is_rejoin_procedure_started;
+static bool is_rejoin_stop_requested;
+static bool is_rejoin_in_progress;
+static uint8_t rejoin_attempt_cnt;
 #if defined CONFIG_ZIGBEE_ROLE_END_DEVICE
-static volatile bool  wait_for_user_input;
-static volatile bool  is_rejoin_start_scheduled;
+static volatile bool wait_for_user_input;
+static volatile bool is_rejoin_start_scheduled;
 #endif
 
+#if defined CONFIG_ZIGBEE_FACTORY_RESET
+static bool factory_reset_done;
```

Comment:
Consider moving this flag to the `factory_reset_context_t`

Reply:
Done